### DISCUSSION TITLE
`chainId` as `bigint`

### DISCUSSION THREAD

**Discussion by 381e35** [wagmi-dev/abitype#209] [Q&A]
Hello, I have been working around signed types/messages and have run into quite a perplexing situation. Right now, the declaration of `TypedDataDomain` is:

```typescript
export type TypedDataDomain = {
  chainId?: number | undefined
  name?: string | undefined
  salt?: ResolvedRegister['BytesType']['outputs'] | undefined
  verifyingContract?: Address | undefined
  version?: string | undefined
}
```

This is fine in most cases as there are not currently any chains with a large `chainId`, however I think there may be a case to treat it as a `uint256` like specified in [EIP-712](https://eips.ethereum.org/EIPS/eip-712#:~:text=uint256%20chainId%20the%20EIP%2D155%20chain%20id.%20The%20user%2Dagent%20should%20refuse%20signing%20if%20it%20does%20not%20match%20the%20currently%20active%20chain.)

Right now, if you are working with signed types you will inevitably find yourself needing to declare the domain type like:

```
export const EIP712_TYPES = {
	EIP712Domain: [
		{ name: 'name', type: 'string' },
		{ name: 'version', type: 'string' },
		{ name: 'chainId', type: 'uint256' },
		{ name: 'verifyingContract', type: 'address' },
		{ name: 'salt', type: 'bytes32' }
	]
} as const
```

Again, we are still fine here. However, as you can see, `chainId` is a uint256 which means it should "really" be a `bigint` (I think) like here https://github.com/wagmi-dev/abitype/blob/00dd694139d995a98b8994ed180e60bcbc5e2947/packages/abitype/src/utils.ts#L66-L75.

I have been working on a library that automatically builds my entire stack, but `domain` here has sent me down a rabbit hole after building everything around the declared `TypedDataDomain`. It's a `uint256`, so I get a `bigint` as my expected type in one piece, but then when it comes to signing my data, it expects a number. Obviously, I could just do `as bigint` but this is spread everywhere throughout my codebase and is in multiple points: the database, API, etc.

Thoughts? I saw it was updated in [this commit](https://github.com/wagmi-dev/abitype/pull/111/commits/c8320da076111c26f1ea4aa0fd5c812442cbe85e) originally, so maybe I am missing something?

In the end, I got past it with a super hacky fix, but still wonder why that was needed and if this is an issue others have experienced?

Is the expected fix to override the export as is suggested [here](https://github.com/wagmi-dev/abitype/discussions/49#discussioncomment-4009201)?

**Comment by Raiden1411** [comment]
I believe that this was specific to something in wagmi but i was not around when this was changed, so I cant really help. @tmm any thoughts?
